### PR TITLE
Implement `get_node_by_path` API

### LINKED ISSUE #67: Implement `get_node_by_path` API

### PR DESCRIPTION
**Pull request by omdxp**
# Description

Resolves #67 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: c/build.c [hunk 1]
```diff
@@ -146,47 +159,105 @@ void prepend_to_data_file()
...
+    for (size_t i = 0; i < kuliya_with_terms_idx; ++i)
+    {
+        __s_kuliya_schema schema = kuliyas_with_terms[i];
+        fprintf(data_file, "\tfree(%s.terms->slots);\n", schema.__varname);
+        fprintf(data_file, "\tfree(%s.terms);\n", schema.__varname);
+    }
+    fprintf(data_file, "}\n");
+
+    // Prepare get node by path API
+    fprintf(data_file, "\nkuliya_schema *__get_node_by_path(const char* path)\n{");
```
Comment (ZibanPirate) [inline c/build.c:206]:
```suggestion
    fprintf(data_file, "\nkuliya_schema *__get_node_by_path(const char *path)\n{");
```

the C/C++ extension in VSCode keeps on formatting it this way, i guess this is the correct formatting then?

File: c/build.c [hunk 2]
```diff
@@ -146,47 +159,105 @@ void prepend_to_data_file()
...
+        kuliya_with_terms_idx++;
     }
-    fprintf(data_file, "};\n");
+
+    // Add condition clause for path value
+    char clause[100];
+    char path_name[strlen(path_value)];
+    strcpy(path_name, path_value);
+    replace_char(path_name, '_', '/');
+    sprintf(clause, "\n\tif (STR_EQ(\"%s\", path))\n\t\treturn &%s;\n", path_name, path_value);
```
Comment (ZibanPirate) [inline c/build.c:258]: shouldn't we use `switch-case` statement instead? maybe the compiler will optimize it better?

**Review by ZibanPirate — CHANGES_REQUESTED** [review]
awesome work! left few comments

File: c/example/main.c [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    printf("\t\"name\": {\n");
+    printf("\t\t\"ar\": \"%s\",\n", schema->name.ar);
+    printf("\t\t\"en\": \"%s\",\n", schema->name.en);
+    printf("\t\t\"fr\": \"%s\"\n", schema->name.fr);
+    printf("\t},\n");
+    printf("\t\"type\": \"%s\"%s\n", match_node_type(schema->type), schema->terms != NULL ? "," : "");
+    if (schema->terms != NULL)
+    {
+        printf("\t\"terms\": {\n");
+        printf("\t\t\"per_year\": %d,\n", schema->terms->per_year);
```
Comment (ZibanPirate) [inline c/example/main.c:48]:
```suggestion
        printf("\t\t\"perYear\": %d,\n", schema->terms->per_year);
```

we should follow the original schema when outputting a JSON

File: c/build.c (hunk 1 above)
Comment (omdxp) [reply c/build.c:206]: right

File: c/build.c (hunk 2 above)
Comment (omdxp) [reply c/build.c:258]: In C you can't compare strings in switch cases, only integers (enums are other variant of integer values)

Comment (ZibanPirate) [reply c/build.c:258]: right, we can then optimize it later via hashmaps

**Review by ZibanPirate — APPROVED** [review]

**Pull request merged**